frontend/src/common/components/FormDialog.js

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { Form } from 'react-final-form';
+
+// Material UI
+import {
+	Button,
+	Dialog,
+	DialogContent,
+	DialogTitle,
+	Grid,
+} from '@material-ui/core';
+
+// Util
+import ValidationService from '../services/validation.service';
+
+function FormDialog({
+	children,
+	close,
+	maxWidth,
+	open,
+	onCancel,
+	onSubmit,
+	primaryButtonLabel,
+	secondaryButtonLabel,
+	title,
+	validationSchema
+}) {
+
+	const generateErrors = (values) => ValidationService(values, validationSchema).errors;
```

Comment:
Whenever you create a form and want to use this common component, you will also need to add a joi validation schema (see mine below) or following docs:

- https://github.com/hapijs/joi